testutils/ocispec.go

```diff
@@ -0,0 +1,279 @@
+package testutils
+
+import (
+	"github.com/docker/libentitlement/types"
+	"github.com/opencontainers/runtime-spec/specs-go"
+	"reflect"
+)
+
+func capListContains(capList []string, capability types.Capability) bool {
+	capStr := string(capability)
+
+	for _, capElt := range capList {
+		if capElt == capStr {
+			return true
+		}
+	}
+
+	return false
+}
+
+// TestSpec is a test OCI struct with a default Seccomp profile
+func TestSpec() *specs.Spec {
+	s := &specs.Spec{
+		Process: specs.Process{
+			Capabilities: &specs.LinuxCapabilities{
+				Bounding:    []string{},
+				Effective:   []string{},
+				Inheritable: []string{},
+				Permitted:   []string{},
+				Ambient:     []string{},
+			},
+		},
+		Linux: &specs.Linux{
+			Seccomp:   &specs.LinuxSeccomp{},
+			Resources: &specs.LinuxResources{},
+			IntelRdt:  &specs.LinuxIntelRdt{},
+		},
+		Windows: &specs.Windows{
+			Resources: &specs.WindowsResources{
+				Memory:  &specs.WindowsMemoryResources{},
+				CPU:     &specs.WindowsCPUResources{},
+				Storage: &specs.WindowsStorageResources{},
+				Network: &specs.WindowsNetworkResources{},
+			},
+		},
+	}
+
+	seccomp, err := getDefaultSeccompProfile()
+	if err != nil {
+		// In case we get an error before seccomp struct is fully updated from decoded json, we empty it manually.
+		s.Linux.Seccomp = &specs.LinuxSeccomp{DefaultAction: specs.ActErrno}
+	} else {
+		s.Linux.Seccomp = seccomp
+	}
+
+	s.Process.Capabilities.Bounding = getDefaultCapList()
+	s.Process.Capabilities.Effective = getDefaultCapList()
+	s.Process.Capabilities.Inheritable = getDefaultCapList()
+	s.Process.Capabilities.Permitted = getDefaultCapList()
+
+	return s
+}
+
+// SyscallArgsMatchSeccompRuleArgs checks that the seccomp rule's args match the provided syscall args
+func SyscallArgsMatchSeccompRuleArgs(syscallArgsRule, syscallArgsSyscall []specs.LinuxSeccompArg) bool {
+	if len(syscallArgsRule) < len(syscallArgsSyscall) ||
+		(len(syscallArgsSyscall) == 0 && len(syscallArgsRule) != 0) {
+		return false
+	}
+
+	for _, syscallArgFromSyscall := range syscallArgsSyscall {
+		found := false
+
+		for _, syscallArgsFromRule := range syscallArgsRule {
+			if reflect.DeepEqual(syscallArgFromSyscall, syscallArgsFromRule) {
+				found = true
+				break
+			}
+		}
+
+		if !found {
+			return false
+		}
+	}
+
+	return true
+}
+
+// MatchSeccompRule checks that the seccomp rule matches the provided syscall and args
+func MatchSeccompRule(seccompRule specs.LinuxSyscall, syscallName string, syscallArgs []specs.LinuxSeccompArg) bool {
+	for _, name := range seccompRule.Names {
+		if name == syscallName {
+			if SyscallArgsMatchSeccompRuleArgs(seccompRule.Args, syscallArgs) {
+				return true
+			}
+		}
+	}
+
+	return false
+}
+
+// SeccompSyscallWithArgsBlocked checks that the provided syscall and args are blocked by the seccomp profile
+func SeccompSyscallWithArgsBlocked(seccompProfile specs.LinuxSeccomp, syscallName types.Syscall, syscallArgs []specs.LinuxSeccompArg) bool {
```

Comment:
naming nit: Can we rename this to indicate that this is checking something? `SeccompSyscallWithArgsBlocked` doesn't tell us if it's a checker. For example, `IsSyscallBlockedBySeccomp` or `IsSyscallWithArgsBlockedBySeccomp` is a more descriptive name.

Reply:
Yeah I was wondering if the "is" was necessary and indeed it seems weird without it. Will fix all the names.